baybe/surrogates/base.py

```diff
@@ -96,47 +95,22 @@ def transform_targets(self, data: pd.DataFrame) -> pd.DataFrame:
             raise ModelNotTrainedError("The model must be trained first.")
         return self._target_transform(data)
 
-    def posterior(self, candidates: pd.DataFrame) -> tuple[Tensor, Tensor]:
+    def posterior(self, candidates: pd.DataFrame) -> Posterior:
         """Evaluate the surrogate model at the given candidate points."""
-        import torch
-
-        # Evaluate the posterior distribution
-        mean, covar = self._posterior(to_tensor(self.transform_inputs(candidates)))
-
-        # Apply covariance transformation for marginal posterior models
-        if not self.joint_posterior:
-            # Convert to tensor containing covariance matrices
-            covar = torch.diag_embed(covar)
-
-        # Add small diagonal variances for numerical stability
-        covar.add_(torch.eye(covar.shape[-1]) * _MIN_VARIANCE)
-
-        return mean, covar
+        return self._posterior(to_tensor(self.transform_inputs(candidates)))
 
     @abstractmethod
-    def _posterior(self, candidates: Tensor) -> tuple[Tensor, Tensor]:
-        """Perform the actual posterior evaluation logic.
-
-        In contrast to its public counterpart
-        :func:`baybe.surrogates.Surrogate.posterior`, no data
-        validation/transformation is carried out but only the raw posterior computation
-        is conducted.
-
-        Note that the public ``posterior`` method *always* returns a full covariance
-        matrix. By contrast, this method may return either a covariance matrix or a
-        tensor of marginal variances, depending on the models ``joint_posterior``
-        flag. The optional conversion to a covariance matrix is handled by the public
-        method.
-
-        See :func:`baybe.surrogates.Surrogate.posterior` for details on the
-        parameters.
+    def _posterior(self, candidates: Tensor) -> Posterior:
+        """Perform the actual posterior evaluation logic."""
 
-        Args:
-            candidates: The candidates.
+    @staticmethod
+    def _get_model_context(searchspace: SearchSpace, objective: Objective) -> Any:
```

Comment:
- like the context way
- dont see need for renaming, but if desired why not `campaign_context` or so
- `Any` is fine, I dont thing we should make a base class uniting all contexts, users should be able to skipt context entirely or make it simple dicts etc depending on what they need

Reply:
Alright, then let's keep it as is. Not a fan of `campaign_context` because it couples the surrogate to campaign in terms of naming for no reason. `model_context` would be fine, but in all these versions `context` is still part of the word, so not really addressing @AVHopp's point. If you are OK, please thumbs up or resolve